### PR TITLE
subgraph buffer

### PR DESCRIPTION
**Pull request by sanchaymittal**
closes: #343

### CODE REVIEW

File: packages/router/src/bindings/contractReader/index.ts [hunk 1]
```diff
@@ -105,13 +138,13 @@ export const handleSingle = async (
         requestContext,
       });
     }
-    const senderReceipt = await txService.getTransactionReceipt(
+    const senderPerpareReceipt = await txService.getTransactionReceipt(
```
Comment (jakekidd) [inline packages/router/src/bindings/contractReader/index.ts:141]: senderPerpare lol

File: packages/router/src/bindings/contractReader/index.ts [hunk 2]
```diff
@@ -72,21 +84,41 @@ export const handleActiveTransactions = async (transactions: ActiveTransaction<a
...
+      transaction.status === CrosschainTransactionStatus.SenderPrepared ||
+      transaction.status === CrosschainTransactionStatus.SenderExpired
+    ) {
+      chainId = transaction.crosschainTx.invariant.receivingChainId;
+    } else {
+      chainId = transaction.crosschainTx.invariant.sendingChainId;
+    }
+
+    handlingTracker.set(transaction.crosschainTx.invariant.transactionId, {
+      blockNumber: 0,
```
Comment (jakekidd) [inline packages/router/src/bindings/contractReader/index.ts:102]: block number 0?

Comment (jakekidd) [reply packages/router/src/bindings/contractReader/index.ts:102]: Oh I see, you set the block number below

If you're going to set a "substitute" or fake initial block number first, can we make it -1 so it's obvious it's just not set yet?

File: packages/router/src/bindings/contractReader/index.ts [hunk 3]
```diff
@@ -72,21 +84,41 @@ export const handleActiveTransactions = async (transactions: ActiveTransaction<a
...
+      chainId,
+    });
+    const res = await handleSingle(transaction, requestContext);
+    if (res) {
+      handlingTracker.set(transaction.crosschainTx.invariant.transactionId, {
+        blockNumber: res.blockNumber,
+        chainId,
+      });
+    } else {
+      handlingTracker.delete(transaction.crosschainTx.invariant.transactionId);
```
Comment (jakekidd) [inline packages/router/src/bindings/contractReader/index.ts:112]: Why are we delaying when there isn't a `res` ?

File: packages/router/src/bindings/contractReader/index.ts (hunk 2 above)
Comment (jakekidd) [reply packages/router/src/bindings/contractReader/index.ts:102]: nit

File: packages/router/src/bindings/contractReader/index.ts (hunk 3 above)
Comment (sanchaymittal) [reply packages/router/src/bindings/contractReader/index.ts:112]: we are delaying but deleting the record if res is undefined.

File: packages/router/src/bindings/contractReader/index.ts (hunk 2 above)
Comment (sanchaymittal) [reply packages/router/src/bindings/contractReader/index.ts:102]: yeah we can do that

File: packages/router/src/bindings/contractReader/index.ts (hunk 3 above)
Comment (jakekidd) [reply packages/router/src/bindings/contractReader/index.ts:112]: I meant deleting lol

Comment (jakekidd) [reply packages/router/src/bindings/contractReader/index.ts:112]: What does `res` being undefined mean exactly?

Comment (sanchaymittal) [reply packages/router/src/bindings/contractReader/index.ts:112]: handleSingle resolves with undefined if it errors or transaction fails

**Review by rhlsthrm — APPROVED** [review]

**Pull request merged**